payjoin/src/uri/url_ext.rs

```diff
@@ -130,6 +145,24 @@ fn set_param(url: &mut Url, prefix: &str, param: &str) {
     url.set_fragment(if fragment.is_empty() { None } else { Some(&fragment) });
 }
 
+#[cfg(feature = "v2")]
+#[derive(Debug)]
+pub enum ParseEndpointFragmentError {
+    LowercaseEndpointFragment,
+}
+
+#[cfg(feature = "v2")]
+impl std::fmt::Display for ParseEndpointFragmentError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use ParseEndpointFragmentError::*;
+
+        match &self {
+            LowercaseEndpointFragment =>
+                write!(f, "Some or all of the endpoint fragment is lowercase"),
+        }
+    }
+}
+
```

Comment:
Since this only has one variant, I'm not convinced we need to complicate it. We can probably just use the one BadEndpoint::FragmentParse variant or rename it BadEndpoint::LowercaseEndpointFragment and simplify by deleting this.